src/couchapps/WorkQueue/lists/workRestrictions.js

```diff
@@ -55,7 +55,11 @@ function(head, req) {
         };
 		
         var ele = row["doc"]["WMCore.WorkQueue.DataStructs.WorkQueueElement.WorkQueueElement"];
-
+        // team shouldn't be empty string or undefined
+        if (teams.length === 0 || !ele["TeamName"]) {
```

Comment:
The line below is performing almost the same checks. How about we either add `|| teams.indexOf(ele["TeamName"]) === -1` to this if and remove the if below or keep the if
below with only the indexOf check?

Reply:
Yes you are right! I can modify that but I need to check why all the unittest failed.